File: src/main/webapp/src/app/models/submission.ts
import {SubmissionStatus} from "./submissionStatusEnum";

export class Submission {
  private _id: number;
  private _taskId: number;
  private _userId: number;
  private _courseEditionId: number;
  private _userFullName: string;
  private _submissionTime: Date;
  private _submissionStatus: SubmissionStatus;
  private _points: number;

  constructor(id: number, taskId: number, userId: number, userFullName: string, courseEditionId: number, submissionTime: Date, submissionStatus: SubmissionStatus, points: number) {
    this._id = id;
    this._taskId = taskId;
    this._userId = userId;
    this._userFullName = userFullName;
    this._courseEditionId = courseEditionId;
    this._submissionTime = submissionTime;
    this._submissionStatus = submissionStatus;
    this._points = points;
  }

  get id(): number {
    return this._id;
  }

  get taskId(): number {
    return this._taskId;
  }

  get userId(): number {
    return this._userId;
  }

  get userFullName(): string {
    return this._userFullName;
  }

  get courseEditionId(): number {
    return this._courseEditionId;
  }

  get submissionTime(): Date {
    return this._submissionTime;
  }

  get points(): number {
    return this._points;
  }

  set points(points: number) {
    this._points = points;
  }

  get submissionStatus(): SubmissionStatus {
    return this._submissionStatus;
  }

  set submissionStatus(submissionStatus: SubmissionStatus) {
    this._submissionStatus = submissionStatus;
  }

  public static fromJSON(jsonObj: any): Submission {
    return new Submission(+jsonObj.id, +jsonObj.taskId, +jsonObj.userId, jsonObj.userFullName, +jsonObj.courseEditionId,
      new Date(jsonObj.submissionTime), jsonObj.submissionStatusType, +jsonObj.points);
  }

  public toJSON() {
    return {
      id: this.id,
      taskId: this.taskId,
      userId: this.userId,
      courseEditionId: this.courseEditionId,
      submissionTime: this.submissionTime,
      points: this.points,
      submissionStatus: this.submissionStatus
    }
  }
}

File: src/main/java/pl/coddlers/core/models/entity/Submission.java
package pl.coddlers.core.models.entity;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.sql.Timestamp;
import java.util.ArrayList;
import java.util.List;


@Entity
@Data
@NoArgsConstructor
@EqualsAndHashCode(exclude = {"user"})
public class Submission {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private Timestamp submissionTime;

    private Integer points;

    private String branchName;

    private String lastCommitHash;

    @ManyToOne(fetch = FetchType.EAGER, targetEntity = SubmissionStatusType.class)
    @JoinColumn(name = "submission_status_type_name")
    private SubmissionStatusType submissionStatusType;

    @ManyToOne(fetch = FetchType.EAGER, targetEntity = StudentLessonRepository.class)
    @JoinColumn(name = "student_lesson_repository_id")
    private StudentLessonRepository studentLessonRepository;

    @OneToMany(mappedBy = "submission", targetEntity = Comment.class)
    private List<Comment> comments = new ArrayList<>();

    @ManyToOne(targetEntity = User.class)
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne(fetch = FetchType.LAZY, targetEntity = CourseEdition.class)
    @JoinColumn(name = "course_edition_id")
    private CourseEdition courseEdition;

    @ManyToOne(targetEntity = Task.class)
    @JoinColumn(name = "task_id")
    private Task task;

    public SubmissionStatusTypeEnum getSubmissionStatusTypeEnum() {
        return SubmissionStatusTypeEnum.getEnumByStatusName(submissionStatusType.getName());
    }
}
